src/Azure.Functions.Cli/Helpers/AzureHelper.cs

```diff
@@ -357,9 +390,75 @@ private static async Task ArmHttpAsync(HttpMethod method, Uri uri, string access
             }
         }
 
-        public static async Task PrintFunctionsInfo(Site functionApp, string accessToken, string managementURL, bool showKeys)
+        private static HttpClient functionAppReadyClient = null;
+
+        internal static async Task WaitUntilFunctionAppReadyAsync(Site functionApp, string accessToken, string managementURL,
+            bool showKeys, HttpMessageHandler messageHandler = null)
         {
-            var functions = await GetFunctions(functionApp, accessToken, managementURL);
+            var masterKey = await GetMasterKeyAsync(functionApp.SiteId, accessToken, managementURL);
```

Comment:
GetMasterKeyAsync can return null. Add check for null before using GetMasterKeyAsync

Reply:
Thanks. Very good point. Done.